modules/serverless-broker-types.adoc

```diff
@@ -1,27 +1,24 @@
 // Module included in the following assemblies:
 //
-// * /serverless/develop/serverless-using-brokers.adoc
+// * /serverless/discover/serverless-brokers.adoc
 
 :_content-type: CONCEPT
 [id="serverless-broker-types_{context}"]
 = Broker types
 
-There are multiple broker implementations available for use with {ServerlessProductName}, each of which have different event delivery guarantees and use different underlying technologies. You can choose the broker implementation when creating a broker by specifying a broker class, otherwise the default broker class is used. The default broker class can be configured by cluster administrators.
-// TO DO: Need to add docs about setting default broker class.
+You can choose the broker implementation when creating a broker by specifying a broker class, otherwise the default broker class is used. The default broker class can be configured by cluster administrators.
 
-[id="serverless-broker-types-channel-based"]
-== Channel-based broker
-
-The channel-based broker implementation internally uses channels for event delivery. Channel-based brokers provide different event delivery guarantees based on the channel implementation a broker instance uses, for example:
+:FeatureName: Kafka broker
+include::snippets/technology-preview.adoc[leveloffset=+2]
 
-* A broker using the `InMemoryChannel` implementation is useful for development and testing purposes, but does not provide adequate event delivery guarantees for production environments.
+[id="serverless-broker-types-default_{context}"]
+== Default broker implementation for development purposes
 
-* A broker using the `KafkaChannel` implementation provides the event delivery guarantees required for a production environment.
+Knative provides a default, channel-based broker implementation. This channel-based broker can be used for development and testing purposes, but does not provide adequate event delivery guarantees for production environments. The default broker is backed by the `InMemoryChannel` channel implementation by default.
 
-[id="serverless-broker-types-kafka"]
-== Kafka broker
+It is possible but not recommended to configure the channel-based broker to be backed by the `KafkaChannel` channel implementation. If you want to use Kafka channels, it is recommended to also use the Kafka broker implementation to reduce network hops.
```

Comment:
> If you want to use Kafka channels, it is recommended to also use the Kafka broker implementation to reduce network hops.

Kafka channel, without the MT-ChannelBroker are OK to use. No extra network hops. Exactly like the Kafka Broker.  One "HTTP Server" for the ingress of the CloudEvents, "persistence" via Kafka, and egress via HTTP client for POSTing the CLoudevents to the Triggers/Subscribers.

The only usage that is actually problematic is the usage of the MT-Channel-Based Broker, with a channel like the Kafka Channel - because of the network hops. In theory the MT-ChannelBroker is also problematic w/ the IMC channel, because it has extra network hops as well, since the MTChannelBroker is POSTing Cloudevents to its underlying channel. using HTTP. regardless if Kafka, IMC or any other channel

Reply:
This section of docs is specifically talking about brokers, not channels without brokers.
I don't think there's a need to call out that it's OK to use kafka channels without certain brokers here; if that's required it should be discussed in the channel docs (please raise another issue for this).
All we are calling out here is as you've mentioned, that we don't recommend using kafka channels with MT broker, to highlight that this broker isn't ideal for Kafka use cases and that they should use the Kafka broker instead.

> In theory the MT-ChannelBroker is also problematic w/ the IMC channel, because it has extra network hops as well, since the MTChannelBroker is POSTing Cloudevents to its underlying channel. using HTTP. regardless if Kafka, IMC or any other channel

That's fine, but again I don't know that this is worth mentioning here - unless we basically want to say there are going to be problems with channel-based brokers no matter what channel you use. I understand that you're trying to stop people from wanting to use MT broker, but we should not be trying to put them off like this from a docs perspective, since all it does is highlight a very negative user experience. If it's paramount that users no longer use the MT broker, it should be retired or deprecated properly from an eng perspective, rather than relying on docs to highlight the negatives.

Reply:
> This section of docs is specifically talking about brokers, not channels without brokers.
I don't think there's a need to call out that it's OK to use kafka channels without certain brokers here; if that's required it should be discussed in the channel docs (please raise another issue for this).

OK, I just saw the section below and was confused:
```
If you want to use Kafka channels, it is recommended to also use the Kafka broker implementation to reduce network hops.
```

After reading it again, in the context of your comment, I think I now got it. Perhaps can we say:

```
If you want to use Kafka channels behind the channel-based broker, it is recommended to also use the Kafka broker implementation instead to reduce network hops.
```